src/box/iproto.cc

```diff
@@ -1411,10 +1411,16 @@ iproto_connection_on_output(ev_loop *loop, struct ev_io *watcher,
 			ev_io_start(loop, &con->output);
```

Comment:
Hm, the test in the first commit passes on my machine even without the fix. Does it fail on your machine? Except when I run it too many times like this `python test-run.py $(yes 6818 | head -100)`, but then it hangs even with the patch. There was a problem with luatest and too many runs in parallel, at least on Mac. Instances are not all killed in the end and it randomly hangs. So the hang probably isn't related to the patch.

Reply:
On my machine it didn't fail either. I slightly tuned the test - now it fails all the time without the patch and passes with the patch. I couldn't reproduce the hang running the test in parallel - looks like a test-run issue after all.

<details><summary>diff</summary>

```diff
diff --git a/test/box-luatest/gh_6818_iproto_resumes_input_if_no_output_test.lua b/test/box-luatest/gh_6818_iproto_resumes_input_if_no_output_test.lua
index 8bf44e18eb5e..d5675061d1bd 100644
--- a/test/box-luatest/gh_6818_iproto_resumes_input_if_no_output_test.lua
+++ b/test/box-luatest/gh_6818_iproto_resumes_input_if_no_output_test.lua
@@ -8,7 +8,7 @@ g.before_all = function()
     g.server = server:new({
         alias = 'master',
         box_cfg = {
-            net_msg_max = 16,
+            net_msg_max = 4,
         }
     })
     g.server:start()
@@ -31,13 +31,13 @@ g.test_iproto_resumes_input_if_no_output = function()
             local conn = net.connect(g.server.net_box_uri)
             conn:watch('foo', function() end)
             for _ = 1, loop_count do
-                conn:eval('return true')
+                conn:eval([[require('fiber').sleep(0.001)]])
             end
             conn:close()
             on_done:put(true)
         end)
     end
     for _ = 1, fiber_count do
-        t.assert(on_done:get())
+        t.assert(on_done:get(10))
     end
 end
```

</details>